BlueprintUI/Sources/Layout/ElementDimensions.swift

```diff
@@ -0,0 +1,51 @@
+import CoreGraphics
+
+/// An element’s size and its alignment guides in its own coordinate space.
+public struct ElementDimensions: Equatable {
+
+    /// The element's width
+    public internal(set) var width: CGFloat
+
+    /// The element's height
+    public internal(set) var height: CGFloat
+
+    private var horizontalGuideValues: [ObjectIdentifier: CGFloat] = [:]
+    private var verticalGuideValues: [ObjectIdentifier: CGFloat] = [:]
+
+    init(size: CGSize) {
+        self.width = size.width
+        self.height = size.height
+    }
+
+    /// Accesses the value of the given guide.
+    public internal(set) subscript(guide: HorizontalAlignment) -> CGFloat {
```

Comment:
Could just be me but not the biggest fan of keypaths here because I always find this less discoverable. At the very least, please document that you need to use keypaths to get at the guide values for this type :)

Reply:
Do you mean subscripts, not keypaths? Or are you talking about the static instances on `HorizontalAlignment` and `VerticalAlignment`?

Reply:
Monday 😅 I did mean subscripts. But thinking about how a func would be shaped, not sure it's better - happy keeping it as a subscript

Reply:
Gotcha. This API is meant to mirror [ViewDimensions](https://developer.apple.com/documentation/swiftui/viewdimensions), but I will do a documentation pass.